src/targets/maven.ts

```diff
@@ -0,0 +1,250 @@
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug('Checking Maven CLI is available...');
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
+    this.logger.debug('Checking GPG is available...');
+    checkExecutableIsPresent('gpg');
+  }
+
+  public async publish(_version: string, _revison: string): Promise<void> {
+    await this.createUserGradlePropsFile();
+    if (isDryRun()) {
+      this.logger.info('[dry-run] Not uploading to Maven.');
+      return;
+    }
+    await this.upload();
+    this.closeAndRelease();
+  }
+
+  /**
+   * Deploys to Maven Central the distribution packages.
+   * Note that after upload, this must be `closeAndRelease`.
+   */
+  public async upload(): Promise<void> {
+    const distributionsDirs = await fs.promises.readdir(
+      this.mavenConfig.distributionsPath
+    );
+    for (const distDir of distributionsDirs) {
+      const moduleName = path.parse(distDir).base;
+      const androidFile = await this.getAndroidDistributionFile(distDir);
+      const targetFile = androidFile
+        ? androidFile
+        : path.join(distDir, `${moduleName}.jar`);
+      const javadocFile = path.join(distDir, `${moduleName}-javadoc.jar`);
+      const sourcesFile = path.join(distDir, `${moduleName}-sources.jar`);
+      const pomFile = path.join(distDir, 'pom-default.xml');
+
+      const command = this.getMavenUploadCmd(
+        targetFile,
+        javadocFile,
+        sourcesFile,
+        pomFile
+      );
+
+      await withRetry(() => {
+        exec(command, (error, _stdout, _stderr) => {
+          if (error) {
+            throw new Error(`Cannot upload ${distDir} to Maven:\n` + error);
+          }
+        });
+        // Not handling an exception forces `withRetry` to try executing the
+        // function again. In the very minute this point has been reached,
+        // there've been no errors running the command, so there's no need to
+        // return any other thing than resolving the promise.
+        return Promise.resolve();
+      });
+    }
+  }
+
+  /**
+   * Finishes the release flow.
+   */
+  public closeAndRelease(): void {
+    const gradleCliAbsPath = path.resolve(this.mavenConfig.gradleCliPath);
+    withRetry(() => {
+      exec(
+        `${gradleCliAbsPath} closeAndReleaseRepository`,
+        (error, _stdout, _stderr) => {
+          if (error) {
+            throw new Error(`Cannot close and release to Maven:\n` + error);
+          }
+        }
+      );
+      // Not handling an exception forces `withRetry` to try executing the
+      // function again. In the very minute this point has been reached,
+      // there've been no errors running the command, so there's no need to
+      // return any other thing than resolving the promise.
+      return Promise.resolve();
+    });
```

Comment:
Pro-tip: if you see yourself repeating a questionable pattern, at least consider encapsulating it in a helper function ;)